logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/SendEditTextMessageUseCase.kt

```diff
@@ -83,32 +83,46 @@ class SendEditTextMessageUseCase internal constructor(
                 newMentions = mentions
             )
             val message = Message.Signaling(
-                id = generatedMessageUuid,
+                id = editedMessageId,
                 content = content,
                 conversationId = conversationId,
                 date = DateTimeUtil.currentIsoDateTimeString(),
                 senderUserId = selfUserId,
                 senderClientId = clientId,
                 status = Message.Status.PENDING
             )
+            // until the edit send is completed and accepted by the backend, we don't change the message id to be able to handle any
+            // incoming edits from other clients that happened in the meantime and already changed the message id
             messageRepository.updateTextMessage(
                 conversationId = message.conversationId,
                 messageContent = content,
-                newMessageId = generatedMessageUuid,
+                newMessageId = originalMessageId,
                 editTimeStamp = message.date
             ).flatMap {
-                messageRepository.updateMessageStatus(
-                    messageStatus = MessageEntity.Status.PENDING,
-                    conversationId = message.conversationId,
-                    messageUuid = generatedMessageUuid
-                )
-            }.flatMap {
-                messageSender.sendMessage(message)
-            }.flatMap {
-                messageRepository.updateMessageStatus(MessageEntity.Status.SENT, conversationId, generatedMessageUuid)
-            }
+                    messageRepository.updateMessageStatus(
+                        messageStatus = MessageEntity.Status.PENDING,
+                        conversationId = message.conversationId,
+                        messageUuid = originalMessageId
+                    )
+                }
+                .flatMap {
+                    messageSender.sendMessage(message)
+                        .flatMap {
+                            // when succeeded, we modify the message id locally
+                            messageRepository.updateTextMessage(
```

Comment:
There's  a race condition here:

1. Client1 sends edit (id: 2) for message1 (inflight)
2. Receives. & process  incoming edit (id: 3) for message1 from client2
3. Client1 edit is accepted by backend
4. We update the id of message1 to 2 

Step 4 is wrong, all other clients will reject client1's edit since there' no matching message with that id.

We should only update the message id to `editedMessageId` here if the previous message id still `originalMessageId`

Reply:
I believe it won't be the case, if the message id has been changed in the meantime then this `updateTextMessage` won't update it since the message with `originalMessageId` doesn't exist anymore in the DB (current message id which needs to be updated is passed in the `messageContent`), so I would just change the `flatMap` to `onSuccess` to not fail the whole process if the `updateTextMessage` doesn't return `Either.Right`. What do you think?

Reply:
I believe it won't be the case, if the message id has been changed in the meantime then this `updateTextMessage` won't update it since the message with `originalMessageId` doesn't exist anymore in the DB (current message id which needs to be updated is passed in the `messageContent`), so I would just change the `flatMap` to `onSuccess` to not fail the whole process if the `updateTextMessage` doesn't return `Either.Right`. What do you think?

Reply:
You're right! Sounds good